R/app.R

```diff
@@ -105,10 +103,17 @@ configure_logger <- function() {
   }
 }
 
-load_main_module <- function() {
+load_main_source <- function() {
+  main <- new.env(parent = globalenv())
+  source(fs::path("app", "main.R"), local = main)
+  main
+}
+
+load_main_box <- function() {
   # Silence "no visible binding" notes raised by `box::use()` on R CMD check.
   app <- NULL
   main <- NULL
+  box::purge_cache()
```

Comment:
Will this run multiple times per session? If user A opens the app, then user B opens the app will `load_main_box` purge all modules even for user A? Or is it once per session (while creating `shinyApp` object for Shiny?

Reply:
Under normal circumstances (no reloading due to changed files) it will run only once per `shiny::runApp()` call.

When Shiny sources `app.R`, `rhino::app()` runs, purges box cache, loads the main module and makes it into a `shinyApp()` object. This object is passed to Shiny and no further calls to `rhino::app()` are made while the app is running (unless `app.R` is modified, in which case Shiny will source it again).